### PR TITLE
[Components with more than 1 service] LPS-187067 Refactor components from frontend-js and frontend-theme modules

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://issues.liferay.com/browse/LPS-187067

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by Kvale1733** [comment]
ci:stop

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
ci:test:startup

**Comment by Kvale1733** [comment]
ci:test:frontend

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

File: modules/apps/frontend-js/frontend-js-importmaps-extender/src/main/java/com/liferay/frontend/js/importmaps/extender/internal/servlet/taglib/JSImportMapsExtenderTopHeadDynamicInclude.java [hunk 1]
```diff
@@ -159,13 +138,42 @@ protected void modified(Map<String, Object> properties) {
...
+		if (_globalImportMapJSONObjects.isEmpty() &&
+			_scopedImportMapJSONObjects.isEmpty()) {
+
+			return true;
+		}
+
+		return false;
+	}
+
+	private String _rebuildImportMaps(JSONFactory jsonFactory) {
```
Comment (rafaprax) [inline modules/apps/frontend-js/frontend-js-importmaps-extender/src/main/java/com/liferay/frontend/js/importmaps/extender/internal/servlet/taglib/JSImportMapsExtenderTopHeadDynamicInclude.java:170]: Hey @MrJohn1911, you don't need this parameter here. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:frontend

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/frontend-js/frontend-js-importmaps-extender/src/main/java/com/liferay/frontend/js/importmaps/extender/internal/servlet/taglib/JSImportMapsExtenderTopHeadDynamicInclude.java [hunk 2]
```diff
@@ -96,53 +99,32 @@ public void register(DynamicIncludeRegistry dynamicIncludeRegistry) {
...
+			_serviceTrackerCustomizer);
 
-			@Override
-			public void unregister() {
-				_scopedImportMapJSONObjects.remove(scope);
-			}
+		_serviceTracker.open();
 
-		};
+		modified();
```
Comment (rafaprax) [inline modules/apps/frontend-js/frontend-js-importmaps-extender/src/main/java/com/liferay/frontend/js/importmaps/extender/internal/servlet/taglib/JSImportMapsExtenderTopHeadDynamicInclude.java:114]: Hey @MrJohn1911, put the `modified` right after the `_bundleContext` initialization.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Review by rafaprax — APPROVED** [review]

**Comment by dantewang** [comment]
See https://github.com/liferay-core-infra/liferay-portal/pull/5339

**Pull request closed without merging**